src/main/java/io/neonbee/data/internal/DataContextImpl.java

```diff
@@ -104,16 +116,24 @@ public DataContextImpl(String correlationId, String bearerToken, JsonObject user
         this(correlationId, null, bearerToken, userPrincipal, data, paths);
     }
 
-    @SuppressWarnings("PMD.ConstructorCallsOverridableMethod")
+    @SuppressWarnings({ "PMD.ConstructorCallsOverridableMethod", "ChainingConstructorIgnoresParameter",
+            "PMD.UnusedFormalParameter" })
     public DataContextImpl(String correlationId, String sessionId, String bearerToken, JsonObject userPrincipal,
             Map<String, Object> data, Deque<DataVerticleCoordinate> paths) {
+        this(correlationId, sessionId, bearerToken, userPrincipal, data, null, paths);
+    }
+
+    @SuppressWarnings({ "PMD.ConstructorCallsOverridableMethod" })
+    public DataContextImpl(String correlationId, String sessionId, String bearerToken, JsonObject userPrincipal,
+            Map<String, Object> data, Map<String, Object> resposneData, Deque<DataVerticleCoordinate> paths) {
         this.correlationId = correlationId;
         this.sessionId = sessionId;
         this.bearerToken = bearerToken;
         // actually create a read only user principal object, so that no one can tamper with the data
         this.userPrincipal = Optional.ofNullable(userPrincipal).map(JsonObject::getMap)
                 .map(Collections::unmodifiableMap).map(JsonObject::new).orElse(null);
         this.setData(data); // create a mutable copy of the map
+        this.responseData = resposneData;
```

Comment:
typo and the resposneData that is passed in might not be mutable, so similar to setData better create a mutable copy?